index.js

```diff
@@ -133,11 +133,13 @@ var createAssertion = function createAssertion(type, message, wrappers, block, m
 	if (descriptors.length === 0) {
 		throw new RangeError(inspect(type) + ' called with no wrappers defined');
 	}
-	var describeMsgs = [];
-	forEach(wrappers, function (wrapper) {
-		checkThis(wrapper);
-		describeMsgs.push(getThisDescription(wrapper));
```

Comment:
Gross, you're right. I'm not a good programmer.

Hmm - then the common case of `this.use(wrapperA, arg1, arg2).use(wrapperB, arg1, arg2).extend('description')` becomes `this.use(wrapperA, arg1, arg2).use(wrapperB, arg1, arg2).extend({ description: 'description' })`, which is slightly more verbose. I'm not sure what you mean about `.use` taking a descriptor; `use` is meant to take an existing plugin, not a descriptor.

OK, so - let's keep your change that modifies `descriptor`, for sure. However, I think that means we should remove the `getThisDescription` and `setThisDescription` helpers entirely?